lib/pg_party/adapter_decorator.rb

```diff
@@ -329,6 +316,27 @@ def add_index_from_options(table_name, name:, type:, algorithm:, using:, columns
               "#{quote_table_name(table_name)} #{using} (#{columns})#{options}"
     end
 
+    def extract_index_options(add_index_options_result)
+      # Rails 6.1 changes the result of #add_index_options
+      index_definition = add_index_options_result.first
+      return add_index_options_result unless index_definition.is_a?(ActiveRecord::ConnectionAdapters::IndexDefinition)
+
+      index_columns = if index_definition.columns.is_a?(String)
+                        index_definition.columns
+                      else
+                        quoted_columns_for_index(index_definition.columns, index_definition.column_options)
+                      end
+
+      [
+        index_definition.name,
+        index_definition.unique ? 'UNIQUE' : index_definition.type,
+        index_columns,
+        index_definition.where ? " WHERE #{index_definition.where}" : nil,
+        add_index_options_result.second, # algorithm option
```

Comment:
One thing I'm curious about... If we don't set an algorithm, what does the index definition look like? Like is there always going to be a second element in this array? If there's not an existing test for that maybe we should add one

Reply:
It will always be an Array with 3 elements. If no algorithm is passed in it will be `nil`. I'm happy to add the test though.

Here's the relevant Rails code: https://github.com/rails/rails/blob/6-1-stable/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb#L1311